pkgs/development/python-modules/mock-open/default.nix

```diff
@@ -1,17 +1,25 @@
-{ lib, buildPythonPackage, fetchFromGitHub, pythonOlder, mock }:
+{ lib, buildPythonPackage, fetchFromGitHub, fetchpatch, pythonOlder, mock }:
 
 buildPythonPackage rec {
   pname = "mock-open";
-  version = "1.3.1";
+  version = "1.4.0";
 
   # no tests in PyPI tarball
   src = fetchFromGitHub {
     owner = "nivbend";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0ikhrhlkl5c7qbigpsv44jw89ws1z7j06gzyg5dh1ki533ifbjm2";
+    sha256 = "0qlz4y8jqxsnmqg03yp9f87rmnjrvmxm5qvm6n1218gm9k5dixbm";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "ascii-only.patch";
+      url = "https://github.com/das-g/mock-open/commit/521ff260da127949fe4aceff1667cba223c5b07b.patch";
+      sha256 = "0ampbhk7kwkn0q5d2h9wrflkr8fji2bybmdck4qdzw1qkslfwwrn";
+    })
+  ];
+
   propagatedBuildInputs = lib.optional (pythonOlder "3.3") mock;
```

Comment:
As this Python version isn't in nixpkgs `master` anymore, should this line be removed?

Reply:
Oh, wait, it'd also be needed for 2.7, so all we could do is replace `(pythonOlder "3.3")` by `isPy27`.

Reply:
If someone's done the work to find out that the actual supported range of versions is, I'd just leave it. Could always have someone trying to use their own custom-built 3.2 they're trying to test something historical against.